enumerable.rb

```diff
@@ -0,0 +1,106 @@
+module Enumerable
+  def my_each
+    return enum_for unless block_given?
+
+    arr = is_a?(Range) ? to_a : self
+    i = 0
+    while i < arr.size
+      yield(arr[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return enum_for unless block_given?
+
+    arr = is_a?(Range) ? to_a : self
+    i = 0
+    while i < arr.size
+      yield(arr[i], i)
+      i += 1
+    end
+    self
+  end
+
+  def my_count
+    return length unless block_given?
+
+    arr = is_a?(Range) ? to_a : self
+    count = 0
+    i = 0
+    while i < arr.size
+      count += 1 if yield(arr[i])
+      i += 1
+    end
+    count
+  end
+
+  def my_select
+    return enum_for unless block_given?
+
+    arr = is_a?(Range) ? to_a : self
+    count = []
+    i = 0
+    while i < arr.size
+      count << arr[i] if yield(arr[i])
+      i += 1
+    end
+    count
+  end
+
+  def my_all?
+    arr = is_a?(Range) ? to_a : self
+    i = 0
+    while i < arr.size
+      yield(arr[i])
+      return false unless yield(arr[i])
+
+      i += 1
+    end
+    true
+  end
```

Comment:
* [ ] should have working `my_all?`  identical and returns the same thing as ruby's [`all?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

#### my_all? 

- [ ] when no block or argument is given returns true only when none of the collection members are false or nil if the mentioned condition is not met it should return false
```ruby     
[nil, false, true, []].my_all? == [nil, false, true, []].all? => false
```
- Yours is giving LocalJumpError
![image](https://user-images.githubusercontent.com/52209067/79677002-6f995f80-81a9-11ea-8529-382c7340b0ff.png)

- Real all?
![image](https://user-images.githubusercontent.com/52209067/77688266-67148700-6f65-11ea-82e5-cbef16444817.png)

- [ ] when a class is passed as an argument returns true only if all of the collection is a member of such class, otherwise, it should return false
```ruby     
[1, 2.5, 'a', 9].my_all?(Integer) == [1, 2.5, 'a', 9].all?(Integer) => false
```
- Yours is retrieving a ArgumentError
![image](https://user-images.githubusercontent.com/52209067/79677015-9192e200-81a9-11ea-93f3-36d170cb3aa3.png)

- Real all?
![image](https://user-images.githubusercontent.com/52209067/77688467-be1a5c00-6f65-11ea-8970-15c9e4f4913f.png)

- [ ] when a Regex is passed as an argument returns true if all of the collection matches the Regex
```ruby     
%w[dog door rod blade].my_all?(/d/) == %w[dog door rod blade].all?(/d/) => true
```
- Yours is retrieving a ArgumentError
![image](https://user-images.githubusercontent.com/52209067/79677025-ae2f1a00-81a9-11ea-937d-2aa43dda47db.png)

- Real all?
![image](https://user-images.githubusercontent.com/52209067/77688741-31bc6900-6f66-11ea-90ec-b6a7cee7e6ca.png)

- [ ] when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern

```ruby     
[3,4,7,11].my_all?(3) == [3,4,7,11].all?(3)  => false 
```
- Yours is retrieving an ArgumentError
![image](https://user-images.githubusercontent.com/52209067/79677032-cacb5200-81a9-11ea-9fc7-c26106048b3d.png)

- Real all?
![image](https://user-images.githubusercontent.com/52209067/77689001-924ba600-6f66-11ea-9b72-210b3bb1d78b.png)

Reply:
Thank you for that amazing review. I have now fix it and its working well.